src/Toolkit/Toolkit/UI/Controls/Bookmarks/Bookmarks.cs

```diff
@@ -0,0 +1,291 @@
+//  *
+//  *  http://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  *   Unless required by applicable law or agreed to in writing, software
+//  *   distributed under the License is distributed on an "AS IS" BASIS,
+//  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  *   See the License for the specific language governing permissions and
+//  *   limitations under the License.
+//  ******************************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.UI.Controls;
+using System.Linq;
+#if NETFX_CORE
+using Windows.UI.Xaml.Controls;
+#elif __IOS__
+using Control = UIKit.UIView;
+#elif __ANDROID__
+using Android.App;
+using Android.Views;
+using Control = Android.Widget.ListView;
+#else
+using System.Windows.Controls;
+#endif
+
+namespace Esri.ArcGISRuntime.Toolkit.UI.Controls
+{
+    /// <summary>
+    /// The Bookmarks view presents bookmarks, either from a list defined by <see cref="BookmarkList" /> or
+    /// the Map or Scene shown in the associated <see cref="GeoView" />.
+    /// </summary>
+    public partial class Bookmarks : Control, INotifyPropertyChanged
+    {
+#if !__ANDROID__
+        public Bookmarks()
+            : base()
+        {
+            Initialize();
+        }
+#endif
+
+        /// <summary>
+        /// Gets or sets the list of bookmarks to display.
+        /// These bookmarks will only be shown if <see cref="PrefersBookmarksList" /> is <code>True</code>.
+        /// Otherwise, the bookmarks from the Map or Scene shown in the associated <see cref="GeoView" /> are displayed.
+        /// </summary>
+        /// <remarks>If set to a <see cref="System.Collections.Specialized.INotifyCollectionChanged" />, the view will be updated with collection changes.</remarks>
+        public IList<Bookmark> BookmarkList
+        {
+            get => BookmarkListImpl;
+            set
+            {
+                BookmarkListImpl = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(BookmarkList)));
+                Refresh();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether bookmarks should be shown from the map/scene or the explicitly set bookmark list.
+        /// When true, the control only shows the bookmarks explicitly set through the <see cref="BookmarkList" /> property.
+        /// Bookmarks from the Map or Scene are ignored, even if the map or scene is changed in the associated <see cref="GeoView" />.
+        /// </summary>
+        public bool PrefersBookmarksList
+        {
+            get => PrefersBookmarkListImpl;
+            set
+            {
+                PrefersBookmarkListImpl = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(PrefersBookmarksList)));
```

Comment:
We only do this on iOS/Android. WPF/UWP we rely on DependencyProperties to do this